File: src/main/java/com/jdjr/crawler/tcpj/exception/GlobalExceptionHandler.java
package com.jdjr.crawler.tcpj.exception;

import com.jdjr.crawler.tcpj.common.enums.SystemCodeEnums;
import com.jdjr.crawler.tcpj.common.exception.AppException;
import com.jdjr.crawler.tcpj.common.result.BasicResult;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;

/**
 * 全局异常处理
 */
@ControllerAdvice
@Slf4j
public class GlobalExceptionHandler {

    /**
     * 其他异常
     */
    @ExceptionHandler(AppException.class)
    @ResponseBody
    public BasicResult appExceptionHandler(AppException e) {
        logger.error("{}", e.getMessage(), e);
        BasicResult result = new BasicResult();
        result.setCode(e.getCode());
        result.setMsg(e.getMsg());
        return result;
    }

    /**
     * 其他异常
     */
    @ExceptionHandler
    @ResponseBody
    public BasicResult allExceptionHandler(Exception e) {
        logger.info("所有异常");
        logger.error("{}", e.getMessage(), e);
        return SystemCodeEnums.ERROR.applyValue();
    }
}

File: spring-security-server/src/main/java/com/go/controller/GlobalExceptionHandler.kt
package com.go.controller

import com.go.common.R
import com.go.common.extension.log
import jakarta.servlet.http.HttpServletRequest
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

/**
 * 全局异常处理
 */
@RestControllerAdvice
class GlobalExceptionHandler {
    /**
     * 系统异常
     */
    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception, request: HttpServletRequest): R<Any?> {
        log.error("系统异常 ${e.message} $e")
        return R.fail()
    }
}